### PR TITLE
Improve cluster checker

### PR DESCRIPTION
**Pull request by miiu96**
- Extended the `cluster-checker` tool

### CODE REVIEW

File: tools/clusters-checker/cmd/checker/flags.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	// defines the path to the config folder
+	configPath = cli.StringFlag{
+		Name:  "config-path",
+		Usage: "The path to the config folder",
+		Value: "./",
+	}
+	checkCounts = cli.BoolFlag{
+		Name:  "check-counts",
+		Usage: "If set, the checker wil verify the counts between clusters",
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/cmd/checker/flags.go:19]: fixed

File: tools/clusters-checker/pkg/checkers/data.go [hunk 2]
```diff
@@ -10,3 +10,8 @@ type generalElasticResponse struct {
 		} `json:"hits"`
 	} `json:"hits"`
 }
+
+type Interval struct {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/pkg/checkers/data.go:14]: added

File: tools/clusters-checker/pkg/checkers/factory.go [hunk 3]
```diff
@@ -39,24 +41,64 @@ func CreateClusterChecker(cfg *config.Config, timestampIndex int, logPrefix stri
...
+		startTimestamp: int(interval.start),
+		stopTimestamp:  int(interval.stop),
 		logPrefix:      logPrefix,
+		onlyIDs:        onlyIDs,
 	}, nil
 }
 
-func CreateMultipleCheckers(cfg *config.Config) ([]*clusterChecker, error) {
-	checkers := make([]*clusterChecker, 0, len(cfg.Compare.IntervalSettings))
+func CreateMultipleCheckers(cfg *config.Config, onlyIDs bool) ([]*clusterChecker, error) {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/pkg/checkers/factory.go:52]: added

**Review by bogdan-rosianu — DISMISSED** [review]

File: tools/clusters-checker/pkg/checkers/factory.go [hunk 4]
```diff
@@ -39,24 +41,65 @@ func CreateClusterChecker(cfg *config.Config, timestampIndex int, logPrefix stri
...
+			start: startTime,
+			stop:  endTime,
+		}}, nil
+	}
+
+	difference := endTime - startTime
+
+	step := difference / numIntervals
+
+	intervals := make([]*Interval, 0)
```
Comment (sstanculeanu) [inline tools/clusters-checker/pkg/checkers/factory.go:89]:
```suggestion
	intervals := make([]*Interval, 0, numIntervals)
```

File: tools/clusters-checker/cmd/checker/flags.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	checkWithTimestamp = cli.BoolFlag{
+		Name:  "check-with-timestamp",
+		Usage: "If set, the checker will verify all the indices from list with timestamp",
+	}
+	checkNoTimestamp = cli.BoolFlag{
+		Name:  "check-no-timestamp",
+		Usage: "If set, the checker will verify the indices from list with no timestamp",
+	}
+	checkOnlyIds = cli.BoolFlag{
+		Name:  "only-ids",
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/cmd/checker/flags.go:30]: done

File: tools/clusters-checker/cmd/checker/flags.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+			" the logs for all packages will have the INFO level, excepting the api package which will receive a DEBUG" +
+			" log level.",
+		Value: "*:" + logger.LogInfo.String(),
+	}
+	logSaveFile = cli.BoolFlag{
+		Name:  "log-save",
+		Usage: "Boolean option for enabling log saving. If set, it will automatically save all the logs into a file.",
+	}
+	// disableAnsiColor defines if the logger subsystem should prevent displaying ANSI colors
+	disableAnsiColor = cli.BoolFlag{
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/cmd/checker/flags.go:46]: changed

File: tools/clusters-checker/cmd/checker/main.go [hunk 7]
```diff
@@ -126,16 +115,21 @@ func checkClusters(ctx *cli.Context) {
 
 	checkWithTimestampFlag := ctx.Bool(checkWithTimestamp.Name)
 	if checkWithTimestampFlag {
-		checkClustersIndexesWithInterval(cfg)
+		checkClustersIndexesWithInterval(cfg, checkOnlyIDs)
 		return
 	}
 
+	if !check.IfNilReflect(fileLogging) {
+		err = fileLogging.Close()
```
[the remark this answers is not in the document]
Comment (miiu96) [reply tools/clusters-checker/cmd/checker/main.go:123]: done

File: tools/clusters-checker/pkg/checkers/factory.go (hunk 4 above)
Comment (miiu96) [reply tools/clusters-checker/pkg/checkers/factory.go:89]: done

**Review by bogdan-rosianu — DISMISSED** [review]

File: tools/clusters-checker/cmd/checker/flags.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+			" log level.",
+		Value: "*:" + logger.LogInfo.String(),
+	}
+	logSaveFile = cli.BoolFlag{
+		Name:  "log-save",
+		Usage: "Boolean option for enabling log saving. If set, it will automatically save all the logs into a file.",
+	}
+	// enableAnsiColor defines if the logger subsystem should displaying ANSI colors
+	enableAnsiColor = cli.BoolFlag{
+		Name:  "disable-ansi-color",
```
Comment (sstanculeanu) [inline tools/clusters-checker/cmd/checker/flags.go:47]:
```suggestion
		Name:  "enable-ansi-color",
```

Comment (miiu96) [reply tools/clusters-checker/cmd/checker/flags.go]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**